### PR TITLE
Extend the color profile PR (#17380)

### PR DESCRIPTION
**Pull request by thomasjo**
### Description of the Change

Extends #17380 by wiring up the dialog prompt for restarting Atom whenever the `core.colorProfile` setting is changed. Since this removes the need for the explicit description remark about restarting Atom, that sentence is also removed.

The third (and optional) change alters the bound functions used when wiring up the change event listeners to arrow functions. We can drop that commit if it's deemed risky, not necessary, or some such.

### Alternate Designs

A more elaborate opt-in mechanism for "prompt for relaunch" was considered, but rejected after taking into consideration the concept of [YAGNI™](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

### Why Should This Be In Core?

For obvious reasons.

### Benefits

Users do not have to manually restart Atom. Some users will likely not read the description and realize they need to manually restart Atom, which would most likely lead to issues being submitted, etc.

### Possible Drawbacks

More code, and complexity?

### Verification Process

Manually tested on Ubuntu 16.04. Works as expected.

### CODE REVIEW

File: src/config-schema.js
Comment (50Wliu) [inline src/config-schema.js]: The relaunch line is included in https://github.com/atom/atom/pull/17464/files#diff-8c99efd21425e5773e52ba4fbb820bf0L574, the other setting that requires a restart.  If this one is removed, then the other should be as well.

Comment (thomasjo) [reply src/config-schema.js]: Oh, I thought I checked that one, but guess I didn't scroll far enough. Good catch, @50Wliu :1st_place_medal: 
Personally I vote for removing it in both places. What do other folks think? /cc @atom/feedback

Comment (daviwil) [reply src/config-schema.js]: I think it's good to inform the user that a restart will be required even if we're going to prompt them.  This will make sure they understand why they're being prompted for a restart.  I'd say leave the message in for the sake of clarity

**Comment by daviwil** [comment]
This change is simple enough that I don't see any problem merging this PR right after I merge the color profile PR.  I think I'd prefer leaving the "restart" language in the setting description though, just for the sake of clarity.

Thanks again for putting this together!

**Comment by daviwil** [comment]
That was ninja-fast ;)

**Review by daviwil — APPROVED** [review]
:bowing_man:

**Comment by daviwil** [comment]
Just waiting on CI to finish before merging this.  Super slow today!

**Pull request merged**
